src/Components/Web.JS/src/Services/ComponentDescriptorDiscovery.ts

```diff
@@ -26,6 +26,34 @@ function discoverServerComponents(document: Document): ServerComponentDescriptor
   return discoveredComponents.sort((a, b): number => a.sequence - b.sequence);
 }
 
+const blazorStateCommentRegularExpression = /\W*Blazor-Component-State:(?<state>[a-zA-Z0-9\+\/=]+)$/;
```

Comment:
I know we have reasons to use HTML comments as a transport mechanism in other cases, but in this particular one (where we know the state doesn't have to be emitted inside an arbitrary element) I wonder if an alternative might be an actual element like `<script type="blazor-component-state">base64goo</script>` or something like that. Then we can find it directly via `document.querySelector` and don't need any regex stuff.

Reply:
That sounds like a good improvement, I was simply following the established pattern that we had. Do you know if there is any issue with using an unknown script type? Would that trigger other HTML compliance tools? I'm hoping that in the proxy case it can be configured to avoid stripping something if necessary.

If you don't think there are potential browser HTML warnings then I'm happy for us to change it, it's easy to change it back later too.

Reply:
updated to use the regex directly